### PR TITLE
feat: Added option to toggle `--explain` flag

### PR DESCRIPTION
**Pull request by mR4smussen**
Eventually this solves https://github.com/flix/vscode-flix/issues/152

This solution works for the compiler used to run the flix programs, but the compiler which compiles on-changes and on-saves seems to use the `--explain` flag as default, and I haven’t been able to find out where this is set.  

This means that this solution works when a program is run and for instance when the :check command is executed, but it’s currently not possible to not get the compiler for on-change and on-safe to not use the `--explain` flag.

Example of the compile on-change without explicitly setting the `--explain` flag:
image

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Maybe because of this: https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/api/Flix.scala#L472

which seems to toggle explain on if there is only one  error? Can you try to test with more than one error?

**Comment by mR4smussen** [comment]
Thanks for taking a look @magnus-madsen. Originally, I had the same thought. The example I have used actually marks two errors, one for each of the `a` functions (I only included one in the image). Originally, I used the example:
```
def main(): Unit \ IO = 
    let x = 123;
    println("Hello World!")
```
Getting the error that `x` is not being used, but like you said, since this only have one error, my solution had no effect on the error message in any of the compilers. But since my current example does work on the “main” compiler, I don’t think this is it.
I'll spent a little more time looking for the course of this in a couple of days.

**Comment by magnus-madsen** [comment]
Ah, I think its on the Flix side... Let me get back.

**Comment by magnus-madsen** [comment]
@mR4smussen Could you try the JAR in https://github.com/flix/flix/pull/5376 ? Thanks!

**Comment by mR4smussen** [comment]
Thanks @magnus-madsen, https://github.com/flix/flix/pull/5376 seems to have fix the problem and I believe we can close this.

**Comment by magnus-madsen** [comment]
You make it non-draft and merge?

**Comment by mR4smussen** [comment]
> "Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests."

I can mark it as ready

**Comment by magnus-madsen** [comment]
Does this work correctly both for what is shown in "output" and what is shown in the terminal?

**Comment by mR4smussen** [comment]
Yes, both compilers should respect the `--explain` flag now

File: package.json [hunk 1]
```diff
@@ -98,6 +98,11 @@
           "default": true,
           "description": "Run the compiler on every change"
         },
+        "flix.explain.enabled": {
+            "type": "boolean",
+            "default": false,
+            "description": "Include suggestions on how to solve errors"
```
Comment (magnus-madsen) [inline package.json:104]: => "Enable detailed compiler error messages."

**Comment by magnus-madsen** [comment]
Perfect, added one comment, and then we can merge.

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**